### PR TITLE
Tn add nn wfst and doc

### PR DESCRIPTION
**Pull request by yzhang123**

### CODE REVIEW

File: docs/source/nlp/text_normalization.rst [hunk 1]
```diff
@@ -17,18 +17,38 @@ system consists of two individual components:
...
 After that, the two trained models can be used to initialize a `DuplexTextNormalizationModel <https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/models/duplex_text_normalization/duplex_tn.py/>`__ that can be used for end-to-end inference.
-An example script for evaluation and inference is provided: `duplex_text_normalization_test.py <https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/duplex_text_normalization/duplex_text_normalization_test.py>`__. The term
-*duplex* refers to the fact that our system can be trained to do both TN and ITN. However, you can also specifically train the system for only one of the tasks.
+An example script for evaluation is provided here: `duplex_text_normalization_test.py <https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/duplex_text_normalization/duplex_text_normalization_test.py>`__. 
+The script for inference of the full pipeline is provided here: `duplex_text_normalization_infer.py <https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/duplex_text_normalization/duplex_text_normalization_infer.py>`__. 
+This script also allows you to interactively run inference in the terminal.
```
Comment (ekmb) [inline docs/source/nlp/text_normalization.rst:22]: `This script also allows you` - > `This script runs inference from a raw text file or an interactive ...`

File: docs/source/nlp/text_normalization.rst [hunk 2]
```diff
@@ -17,18 +17,38 @@ system consists of two individual components:
...
+^^^^^^^^^^^^^^^^
+
+.. list-table:: *Pretrained Models*
+   :widths: 5 10
+   :header-rows: 1
+
+   * - Model
+     - Pretrained Checkpoint
+   * - neural_text_normalization_t5
+     - https://ngc.nvidia.com/catalog/models/nvidia:nemo:neural_text_normalization_t5
```
Comment (ekmb) [inline docs/source/nlp/text_normalization.rst:41]: Is this chpt compatible with all the changes?

File: docs/source/nlp/text_normalization.rst [hunk 3]
```diff
@@ -42,18 +62,36 @@ and ``test.tsv``.
     DATE	2005	two thousand five
     PLAIN	and	<self>
     DATE	2008	two thousand eight
-    PUNCT	.	sil
+    PUNCT	.	<self>
```
Comment (ekmb) [inline docs/source/nlp/text_normalization.rst:65]: could you add a note on why `sil` is converted to `self` and at what stage?

File: docs/source/nlp/text_normalization.rst [hunk 4]
```diff
@@ -42,18 +62,36 @@ and ``test.tsv``.
...
+
+Data preprocessing
+-----------
+
+Processing scripts can be found in the same folder. Right now we only provide scripts for English text normalization, see `data/en/data_preprocessing.py <https://github.com/NVIDIA/NeMo/blob/main/examples/nlp/duplex_text_normalization/data/en/data_preprocessing.py>`__.
+The details can be found at the top of the scripts.
+The purpose of the preprocessing scripts is to standardize the format in order to help with model training.
+For text normalization it is crucial to avoid unrecoverable errors, which are linguistically coherent and not semantic preserving. 
+We noticed that due to data scarcity the model struggles verbalizing long numbers correctly, so we changed the ground truth for long numbers to digit by digit verbalization.
+We also ignore certain semiotic classes from neural verbalization, e.g. `ELECTRONIC` or `WHITELIST`. Instead we label urls/email addresses and abbreviations as plain tokens, and handle it separately with WFST-based grammars, see :ref:`inference_text_normalization`.
```
Comment (ekmb) [inline docs/source/nlp/text_normalization.rst:81]: please add smth like. `WHITELIST`, i.e. VERABUTIM and LETTER in the original dataset

File: docs/source/nlp/text_normalization.rst (hunk 2 above)
Comment (yzhang123) [reply docs/source/nlp/text_normalization.rst:41]: no, i will update the new ckpts and this documentation

File: docs/source/nlp/text_normalization.rst (hunk 3 above)
Comment (yzhang123) [reply docs/source/nlp/text_normalization.rst:65]: ok will add in preprocessing stage

**Review by ekmb — APPROVED** [review]

**Pull request merged**